facebook/delphiFacebook/R/contingency_variables.R

```diff
@@ -406,7 +504,402 @@ create_derivative_columns <- function(df) {
+  
+  # concerned you would experience a side effect: very concerned OR moderately concerned
+  # V9
+  if ("b_concerned_sideeffects" %in% names(df)) {
+    df$b_worried_vaccine_sideeffects <- as.numeric(df$b_concerned_sideeffects == 1)
+  } else {
+    df$b_worried_vaccine_sideeffects <- NA_real_
+  }
+  
+  # ((very concerned OR moderately concerned) AND (no, probably not OR no definitely not))
+  # V3, V9
+  if (all(c("b_worried_vaccine_sideeffects", "b_hesitant_vaccine") %in% names(df))) {
+    df$b_hesitant_worried_vaccine_sideeffects <- as.numeric(
+      df$b_hesitant_vaccine == 1 & df$b_worried_vaccine_sideeffects == 1 
+    )
+  } else {
+    df$b_hesitant_worried_vaccine_sideeffects <- NA_real_
+  }
+  
+  # more likely to get vaccinated if recommended by friends & family among those who are hesitant
+  # ((more likely) AND (no, probably not OR no definitely not))
+  # V3, V4
+  if (all(c("b_hesitant_vaccine", "b_vaccine_likely_friends") %in% names(df))) {
+    df$b_hesitant_vaccine_likely_friends <- as.numeric(
+      df$b_hesitant_vaccine == 1 & df$b_vaccine_likely_friends == 1
+    )
+  } else {
+    df$b_hesitant_vaccine_likely_friends <- NA_real_
+  }
+  
+  # more likely to get vaccinated if recommended by local healthcare workers
+  # among those who are hesitant
+  if (all(c("b_hesitant_vaccine", "b_vaccine_likely_local_health") %in% names(df))) {
+    df$b_hesitant_vaccine_likely_local_health <- as.numeric(
+      df$b_hesitant_vaccine == 1 & df$b_vaccine_likely_local_health == 1
+    )
+  } else {
+    df$b_hesitant_vaccine_likely_local_health <- NA_real_
+  }
+  
+  # more likely to get vaccinated if recommended by WHO among those who are
+  # hesitant
+  if (all(c("b_hesitant_vaccine", "b_vaccine_likely_who") %in% names(df))) {
+    df$b_hesitant_vaccine_likely_who <- as.numeric(
+      df$b_hesitant_vaccine == 1 & df$b_vaccine_likely_who == 1
+    )
+  } else {
+    df$b_hesitant_vaccine_likely_who <- NA_real_
+  }
+  
+  # more likely to get vaccinated if recommended by government health officials
+  # among those who are hesitant
+  if (all(c("b_hesitant_vaccine", "b_vaccine_likely_govt_health") %in% names(df))) {
+    df$b_hesitant_vaccine_likely_govt <- as.numeric(
+      df$b_hesitant_vaccine == 1 & df$b_vaccine_likely_govt_health == 1
+    )
+  } else {
+    df$b_hesitant_vaccine_likely_govt <- NA_real_
+  }
+  
+  # more likely to get vaccinated if recommended by politicians among those who
+  # are hesitant
+  if (all(c("b_hesitant_vaccine", "b_vaccine_likely_politicians") %in% names(df))) {
+    df$b_hesitant_vaccine_likely_politicians <- as.numeric(
+      df$b_hesitant_vaccine == 1 & df$b_vaccine_likely_politicians == 1
+    )
+  } else {
+    df$b_hesitant_vaccine_likely_politicians <- NA_real_
+  }
+  
+  if (all(c("b_hesitant_vaccine", "b_vaccine_likely_doctors") %in% names(df))) {
+    df$b_hesitant_vaccine_likely_doctors <- as.numeric(
+      df$b_hesitant_vaccine == 1 & df$b_vaccine_likely_doctors == 1
```

Comment:
This style of "and"ing doesn't give the right result, because it shortcircuits when one of the variables is False. For example, 1 & NA = NA, but 0 & NA = 0 and NA & 0 = 0. So we end up overcounting the Falses.

Specifically, `b_hesitant_vaccine_likely_doctors` should be True for people who said they're more likely to get vaccinated if recommended by their doctor *among* people who are hesitant. So it requires a non-missing response to `b_hesitant_vaccine` which isn't checked here.

This applies to all the other "hesitant and" variables as well.

Reply:
Done.